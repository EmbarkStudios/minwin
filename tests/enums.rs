mod utils;
use utils::*;

// Ensures we can emit some but not all of a simple enum's values
test!(
    simple,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Foundation.GENERIC_READ",
        "Windows.Win32.Foundation.GENERIC_WRITE",
        "Windows.Win32.Foundation.GENERIC_ALL",
        "Windows.Win32.Storage.FileSystem.FILE_TYPE_DISK",
        "Windows.Win32.Storage.FileSystem.GetFileType",
        "Windows.Win32.Storage.FileSystem.ClfsMgmtLogWriteNotification",
        "Windows.Win32.Storage.FileSystem.ReadLogNotification",
    ]
);

// Ensures we can emit enums that use core types instead of a typedef
test!(
    core,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.Foundation.S_OK"]
);

// Ensures we can emit the enum type alias, even if no actual enum variants
// are requested
test!(
    always_emits_alias,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.UI.Input.Pointer.GetPointerTouchInfo"]
);

// Ensures that enums that clash with handle types are supported
test!(
    handle_and_enum,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.UI.WindowsAndMessaging.SetWindowPos",
        "Windows.Win32.UI.WindowsAndMessaging.GetMessageW",
        "Windows.Win32.UI.WindowsAndMessaging.HWND_NOTOPMOST",
    ]
);
