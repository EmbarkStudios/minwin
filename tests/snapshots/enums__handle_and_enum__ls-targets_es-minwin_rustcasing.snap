---
source: tests/enums.rs
expression: bo.bindings
---
#![allow(
    non_snake_case,
    non_upper_case_globals,
    non_camel_case_types,
    clippy::upper_case_acronyms
)]
::windows_targets::link!(
    "user32.dll" "system" "GetMessageW" fn get_message_w(lp_msg : * mut Msg, h_wnd :
    Hwnd, w_msg_filter_min : u32, w_msg_filter_max : u32) -> Bool
);
::windows_targets::link!(
    "user32.dll" "system" "SetWindowPos" fn set_window_pos(h_wnd : Hwnd,
    h_wnd_insert_after : Hwnd, x : i32, y : i32, cx : i32, cy : i32, u_flags :
    SetWindowPosFlags::Enum) -> Bool
);
pub type Bool = i32;
pub type Hwnd = isize;
pub const HwndNotopmost: Hwnd = -2;
pub type Lparam = isize;
#[repr(C)]
pub struct Msg {
    pub hwnd: Hwnd,
    pub message: u32,
    pub w_param: Wparam,
    pub l_param: Lparam,
    pub time: u32,
    pub pt: Point,
}
#[repr(C)]
pub struct Point {
    pub x: i32,
    pub y: i32,
}
pub mod SetWindowPosFlags {
    pub type Enum = u32;
}
pub type Wparam = usize;

